### PR TITLE
Tqdm and small compile

### PR DESCRIPTION
**Pull request by yallup**
A small quality of life request in a progress bar for training. 

A second small ( but potentially breaking change?) in wrapping the train step in a jit wrapper. Naively gives a factor 2 improvement in run time on CPU and GPU as far as I can see, but may need more advanced checking on other hardware.

More broadly I tried putting the training on a GPU and saw little speed up, I started digging through the code and I think there are quite a lot of conversions to numpy and non-TF operations. So at the moment there isn't much gain from GPU usage here, but the JIT compilation still looks advantageous.

### CODE REVIEW

**Comment by htjb** [comment]
Ahh thanks David! Both of these things looks super helpful. I did play around with jit compile a bit as I had seen it lead to improvements with some other work I had been doing but just hadn't got around to implementing it here. Should help solve #20. 

This shouldn't be a breaking change but we will have to bump the version number up to 0.6.0.

Need to import tqdm I think.

**Comment by htjb** [comment]
Hi @yallup, have you had a chance to look at this? Be great to get these features merged in. Thanks!

**Comment by htjb** [comment]
I think it's because tensorflow is trying to create the function multiple times... in the for loop

```python
for i in range(1,len(self.theta)):
        self.maf[i] = self._training(self.theta[i],
                                     self.sample_weights[i],
                                     self.maf[i], self.theta_min[i],
                                     self.theta_max[i])
```

which I can validate by running

```python
self.maf[0] = self._training(self.theta[0],
                                 self.sample_weights[0],
                                 self.maf[0], self.theta_min[0],
                                 self.theta_max[0])
self.maf[1] = self._training(self.theta[1],
                                 self.sample_weights[1],
                                 self.maf[1], self.theta_min[1],
                                 self.theta_max[1])
```
which gives the same error when it gets to self.maf[1].

**Comment by htjb** [comment]
@yallup I've been trying to resolve this issue but couldn't do it. I'm keen to take advantage of the tqdm addition though so I have removed the `@tf.function(jit_compile=True)` decorator on the training step for now.

I will come back to this when I look at #31 again. If the tests pass could you please squash and merge. Thanks!

**Comment by yallup** [comment]
Totally agree on putting a pin in the speedup stuff, it was just a little thing I was trying. I got a bit caught up trying out some other bits such as reducing translations between tf and numpy operations as that might be a bigger speedup in the long run, one to put a pin in and revisit I think!

**Comment by htjb** [comment]
No worries! Thanks for having a look at it. I started the process of rewriting the MAF class line by line this evening to try and eliminate some of the back and forth between numpy/tensorflow and to add in some tf.function decorators where possible. Seems to be going well and hopefully that will help with run time issues. I'll put a PR in at some point.
